docs/readmes/proposals/p010_vendor_neutral_dp.md

```diff
@@ -416,7 +456,250 @@ scope of work.
+
+The Domain Proxy will be split into separate services in separate containers.
+
+Every service will work in HA mode to provide service even in failover scenarios.
+
+Components such as database and queue will use cluster solutions
+delivered by database and queue providers.
+
+All other components will be stateless Python based services
+which do not need synchronization between them
+so it is not vital to create sync mechanisms.
+Python scripts will take over processing of data
+when one of the containers fail.
+When a container that processes requests received from RabbitMQ fails
+then RabbitMQ will not receive confirmation of received requests from the queue
+and will queue requests again.
+The only state will be kept in the database.
+
+## Batching of messages
+
+Batching of messages allows to limit the number of connections to the SAS servers
+which is useful in the case when a number of CBSDs send requests at once
+(e.g. heartbeat requests are sent continuously).
+Batching it limits a chance of failure on sending requests to the SAS,
+because SAS has to process only a minimum number of requests
+instead of numerous requests sent one by one.
+In large infrastructures it may reduce the cost of SAS usage
+and limit the chance of running out of requests per given time
+(for Google SAS server it is 50,000 requests per day and 10 queries per second).
+
+Keeping the state is not necessary on the early stage of the project
+when Domain Proxy supports only Passive Mode
+where requests only pass through the DP.
+However, on Active Mode it is necessary to keep the CBSDs objects in the database
+as components such as UI monitor registered CBSDs in the Domain Proxy
+and allow to edit attributes of the CBSDs.
+Using stateless solutions would require to send requests to the SAS server
+or CBSDs itself which is not a good design
+considering that it may generate additional costs and add additional traffic
+between Domain Proxy and other components.
+Active Mode with state in the database allows to add additional attributes
+not recognized by SAS or CBSD, but useful for the Domain Proxy usage
+(labels or additional descriptions for UI purposes such as sorting by groups of CBSDs).
+
+## Architectual Discussion
+
+This section contains various technical aspects of the solution that are not solidified and require further investigation
+
+### Relational database
+
+Since there is a planned UI and additionally we need to store basic,
+well defined information about CBSD we suggest to reuse an already existing MySQL / PostgreSQL database
+from the Magma Orchestrator component (with a new schema).
+At the very beginning the database usage would be limited to:
+* storing records for CBSDs configuration (name, location, radio parameters, etc.)
+* keeping track of grants limiting redundant CBSD - SAS connections
+* storing information about operational mode - Active or Passive - per CBSD/DP
+* user administration & access rights within DP system
```

Comment:
Can you elaborate on the configuration models which might be used by configuration controller ?

Reply:
@karthiksubraveti 
Not sure if I understand the question.
The Configuration Controller is responsible for managing CBSD configuration, not Domain Proxy configuration. Maybe that is something we did not outline clearly enough.

The only parameter relevant for the Configuration Controller is the Active or Passive work mode. Come to think of it, most likely a relational database would not be a good place for it (a more appropriate service would be some kind of a key-value store).